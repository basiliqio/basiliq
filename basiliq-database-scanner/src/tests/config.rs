use super::*;

async fn apply_migrations(transaction: &mut sqlx::Transaction<'_, sqlx::Postgres>) {
    sqlx::query!(
        r#"
		CREATE TABLE peoples(
			id			UUID PRIMARY KEY,
			name		TEXT NOT NULL
		);
	"#
    )
    .execute(&mut *transaction)
    .await
    .unwrap();
    sqlx::query!(
        r#"
		CREATE TABLE movies(
			id			UUID PRIMARY KEY,
			title		TEXT NOT NULL
		);
		"#
    )
    .execute(&mut *transaction)
    .await
    .unwrap();
    sqlx::query!(
        r#"
		CREATE TABLE movies_staff(
			id			UUID PRIMARY KEY,
			role		TEXT NOT NULL,
			person		UUID NOT NULL REFERENCES peoples(id) ON DELETE CASCADE,
			movies		UUID NOT NULL REFERENCES movies(id) ON DELETE CASCADE
		);
	"#
    )
    .execute(&mut *transaction)
    .await
    .unwrap();
}

#[ciboulette2postgres_test]
async fn config_default(mut transaction: sqlx::Transaction<'_, sqlx::Postgres>) {
    apply_migrations(&mut transaction).await;
    let raw_table = BasiliqDbScannedTable::scan_db(&mut transaction)
        .await
        .unwrap();
    let builder = BasiliqStoreBuilder::new(raw_table);
    let config = BasiliqStoreConfig::from(&builder);
    insta::assert_json_snapshot!(config);
}
