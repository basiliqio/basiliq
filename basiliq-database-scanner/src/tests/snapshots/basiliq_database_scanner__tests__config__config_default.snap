---
source: basiliq-database-scanner/src/tests/config.rs
expression: config

---
{
  "resources": {
    "public__movies": {
      "target": {
        "schema_name": "public",
        "table_name": "movies"
      },
      "enabled": true,
      "relationships": {
        "movies_staff": {
          "target": {
            "schema_name": "public",
            "table_name": "movies_staff"
          },
          "enabled": true,
          "field": "movies"
        },
        "peoples": {
          "target": {
            "schema_name": "public",
            "table_name": "peoples"
          },
          "through": {
            "schema_name": "public",
            "table_name": "movies_staff",
            "field": "person"
          },
          "enabled": true,
          "field": "id"
        }
      }
    },
    "public__movies_staff": {
      "target": {
        "schema_name": "public",
        "table_name": "movies_staff"
      },
      "enabled": true,
      "relationships": {
        "movies": {
          "target": {
            "schema_name": "public",
            "table_name": "movies"
          },
          "enabled": true,
          "field": "id"
        },
        "peoples": {
          "target": {
            "schema_name": "public",
            "table_name": "peoples"
          },
          "enabled": true,
          "field": "id"
        }
      }
    },
    "public__peoples": {
      "target": {
        "schema_name": "public",
        "table_name": "peoples"
      },
      "enabled": true,
      "relationships": {
        "movies": {
          "target": {
            "schema_name": "public",
            "table_name": "movies"
          },
          "through": {
            "schema_name": "public",
            "table_name": "movies_staff",
            "field": "movies"
          },
          "enabled": true,
          "field": "id"
        },
        "movies_staff": {
          "target": {
            "schema_name": "public",
            "table_name": "movies_staff"
          },
          "enabled": true,
          "field": "person"
        }
      }
    }
  }
}
